qiskit/quantum_info/primitives/estimator/base_estimator.py

```diff
@@ -0,0 +1,158 @@
+from ..results import EstimatorResult
+from ..results.base_result import BaseResult
+from ..utils import Group, init_circuit, init_observable
+
+
+class BaseEstimator(BasePrimitive, ABC):
+    """
+    Expectation Value class
+    """
+
+    def __init__(
+        self,
+        circuits: Union[QuantumCircuit, list[Union[QuantumCircuit]]],
+        observables: Union[BaseOperator, PauliSumOp, list[Union[BaseOperator, PauliSumOp]]],
+        backend: Union[Backend, BackendWrapper],
+        transpile_options: Optional[dict] = None,
+        bound_pass_manager: Optional[PassManager] = None,
+    ):
+        """ """
+        super().__init__(
+            backend=backend,
+            transpile_options=transpile_options,
+            bound_pass_manager=bound_pass_manager,
+        )
+
+        # Initialize quantum circuits
+        if isinstance(circuits, list):
+            self._circuits = [init_circuit(circuit) for circuit in circuits]
+        else:
+            self._circuits = [init_circuit(circuits)]
+
+        # Initialize observables
+        if isinstance(observables, list):
+            self._observables = [init_observable(observable) for observable in observables]
+        else:
+            self._observables = [init_observable(observables)]
+
+        self._grouping = [
+            Group(i, j) for i in range(len(self._circuits)) for j in range(len(self._observables))
+        ]
+        self._transpiled_circuits_cache: dict[tuple[Group], list[QuantumCircuit]] = {}
+
+    @property
+    def circuits(self) -> list[QuantumCircuit]:
+        """Quantum Circuits that represents quantum states.
+
+        Returns:
+            quantum states
+        """
+        return self._circuits
+
+    @property
+    def observables(self) -> list[SparsePauliOp]:
+        """
+        SparsePauliOp that represents observable
+
+        Returns:
+            observable
+        """
+        return self._observables
+
+    @property
+    def grouping(self) -> list[Group]:
+        """
+        Group of circuits and observables.
+
+        Returns:
+            grouping
+        """
+        return self._grouping
+
+    def set_transpile_options(self, **fields) -> BaseEstimator:
+        """Set the transpiler options for transpiler.
+
+        Args:
+            fields: The fields to update the options
+        Returns:
+            self
+        """
+        self._transpiled_circuits = None
+        self._transpiled_circuits_cache = {}
+        super().set_transpile_options(**fields)
+        return self
+
+    @property
+    def preprocessed_circuits(
+        self,
+    ) -> PreprocessedCircuits:
+        """
+        Transpiled quantum circuits produced by preprocessing
+
+        Returns:
+            List of the transpiled quantum circuit
+        """
+        self._preprocessed_circuits = self._preprocessing(self.circuits, self.observables)
+        return super().preprocessed_circuits
+
+    # pylint: disable=arguments-differ
```

Comment:
Do we need to define run() in `BasePrimitive` if we are going to override the signature in the primitives themselves?

Reply:
Thanks. It is defined in the base class side in order to use `super().run` instead of repeating the same process and to define `__call__` in the base class. I believe that there is a pattern in Qiskit https://github.com/Qiskit/qiskit-terra/search?q=arguments-differ. But I'm not that strongly opposed to it, so either is fine.

Reply:
This makes sense, thanks